index.js

```diff
@@ -6,71 +6,78 @@
 const http = require('http');
 const EventEmitter = require('events');
 
-const dflt = function(current, ifUndefined) {
-	return typeof current === 'undefined' ? ifUndefined : current;
+function setValueOrDefault (value, defaultValue) {
+	return typeof value === 'undefined' ? defaultValue : value;
 };
 
-module.exports = function(opts) {
-	opts = dflt(opts, {});
-	opts.port = dflt(opts.port, 3001);
-	opts.readiness = dflt(opts.readiness, {});
-	opts.liveness = dflt(opts.liveness, {});
-	opts.readiness.path = dflt(opts.readiness.path, '/ready');
-	opts.readiness.checker = dflt(opts.readiness.checker, function (next) { return next(); });
-	opts.readiness.checkerTimeoutMs = dflt(opts.readiness.checkerTimeoutMs, 20000);
-	opts.liveness.path = dflt(opts.liveness.path, '/live');
-	opts.liveness.checker = dflt(opts.liveness.checker, function (next) { return next(); });
-	opts.liveness.checkerTimeoutMs = dflt(opts.liveness.checkerTimeoutMs, 20000);
-
-	if (typeof opts.liveness.checker !== 'function' || typeof opts.readiness.checker !== 'function') {
-		throw new Error('Can not create middleware checker option must be a function');
+function initOptions(opts) {
+	opts = setValueOrDefault(opts, {});
+	opts.port = setValueOrDefault(opts.port, 3001);
+	opts.readiness = setValueOrDefault(opts.readiness, {});
+	opts.liveness = setValueOrDefault(opts.liveness, {});
+	opts.readiness.path = setValueOrDefault(opts.readiness.path, '/ready');
+	opts.readiness.createChecker = setValueOrDefault(opts.readiness.createChecker, function () { return opts.readiness.checker; });
+	opts.readiness.checker = setValueOrDefault(opts.readiness.checker, function (next) { return next(); });
+	opts.readiness.checkerTimeoutMs = setValueOrDefault(opts.readiness.checkerTimeoutMs, 20000);
+	opts.liveness.path = setValueOrDefault(opts.liveness.path, '/live');
+	opts.liveness.createChecker = setValueOrDefault(opts.liveness.createChecker, function () { return opts.liveness.checker; });
+	opts.liveness.checker = setValueOrDefault(opts.liveness.checker, function (next) { return next(); });
+	opts.liveness.checkerTimeoutMs = setValueOrDefault(opts.liveness.checkerTimeoutMs, 20000);
+};
+
+function initChecker(checkersInfo, servicePath, checker, timeoutMs) {
```

Comment:
This would rather be `initCheckersInfo` as the checker is already created and "initialized".
I would also rather name it `initCheckersMap` as the goal is to map each `servicePath` to the right checker (and also rename `checkersInfo` as `checkersMap`).

Reply:
The checkersInfo contains the checker and its timeout. The name checkersInfo is not the most accurate or beautiful but checkers only didn't seem right to me either.

Reply:
Replaced by probeMap.

Reply:
Fixed in cddfcf6